sources/CMLREAD

```diff
@@ -237,38 +240,41 @@ Copyright (c) 1985, 1986, 1987, 1988, 1990, 1993 by Venue & Xerox Corporation.
 
 (DECLARE%: EVAL@COMPILE
 
-(DATATYPE READER-ENVIRONMENT (REPACKAGE REREADTABLE REBASE RESPEC))
+(DATATYPE READER-ENVIRONMENT (REPACKAGE REREADTABLE REBASE NIL REFORMAT))
 )
 
-(/DECLAREDATATYPE 'READER-ENVIRONMENT '(POINTER POINTER POINTER POINTER)
+(/DECLAREDATATYPE 'READER-ENVIRONMENT '(POINTER POINTER POINTER POINTER POINTER)
        '((READER-ENVIRONMENT 0 POINTER)
          (READER-ENVIRONMENT 2 POINTER)
          (READER-ENVIRONMENT 4 POINTER)
-         (READER-ENVIRONMENT 6 POINTER))
-       '8)
+         (READER-ENVIRONMENT 6 POINTER)
+         (READER-ENVIRONMENT 8 POINTER))
+       '10)
 
 (DEFMACRO WITH-READER-ENVIRONMENT (ENV . BODY)
    `((CL:LAMBDA (E)
             (LET ((*PACKAGE* (ffetch (READER-ENVIRONMENT REPACKAGE) of E))
                   (*READTABLE* (ffetch (READER-ENVIRONMENT REREADTABLE) of E))
                   (*READ-BASE* (ffetch (READER-ENVIRONMENT REBASE) of E))
-                  (*PRINT-BASE* (ffetch (READER-ENVIRONMENT REBASE) of E)))
+                  (*PRINT-BASE* (ffetch (READER-ENVIRONMENT REBASE) of E))
+                  (*EXTERNALFORMAT* (ffetch (READER-ENVIRONMENT REFORMAT) of E)))
```

Comment:
where are the other variables declared SPECIAL? Why just the external format? Is this really at the right level? The other elements don't depend / vary when you change XCCS and UTF-8... they're EXTERNAL-FORMAT independent

Reply:
Issue #371